sdk/data/azappconfig/internal/generated/custom_client.go

```diff
@@ -21,3 +24,16 @@ func NewAzureAppConfigurationClient(endpoint string, client *azcore.Client) *Azu
 func (a *AzureAppConfigurationClient) Tracer() tracing.Tracer {
 	return a.internal.Tracer()
 }
+
+func NewCreateSnapshotPoller[T any](ctx context.Context, client *AzureAppConfigurationClient, name string, entity Snapshot, options *AzureAppConfigurationClientBeginCreateSnapshotOptions) (*runtime.Poller[T], error) {
```

Comment:
You can make this a method on `*AzureAppConfigurationClient` and remove the `client` parameter.

Reply:
Discussed offline. Method is generic